src/internal/EnhancedButton.js

```diff
@@ -318,12 +327,12 @@ class EnhancedButton extends Component {
     };
     const buttonChildren = this.createButtonChildren();
 
-    // Provides backward compatibity. Added to support wrapping around <a> element.
-    const targetLinkElement = buttonProps.hasOwnProperty('href') ? 'a' : 'span';
+    // Provides backwards compatibility. Added to support wrapping around <a> element.
+    const targetLinkElement = href ? 'a' : 'span';
 
     return React.isValidElement(containerElement) ?
       React.cloneElement(containerElement, buttonProps, buttonChildren) :
-      React.createElement(linkButton ? targetLinkElement : containerElement, buttonProps, buttonChildren);
+      React.createElement((href || listItem) ? targetLinkElement : containerElement, buttonProps, buttonChildren);
```

Comment:
Yeah, this is smelly. We need a better solution. 

Refresh my memory -- what's the goal/purpose of the prop?